base/nullable.jl

```diff
@@ -247,3 +247,45 @@ function hash(x::Nullable, h::UInt)
         return hash(x.value, h + nullablehash_seed)
     end
 end
+
+# higher-order functions
+"""
+    filter(p, x::Nullable)
+
+Return null if either `x` is null or `p(get(x))` is false, and `x` otherwise.
+"""
+function filter{T}(p, x::Nullable{T})
+    if isnull(x) || p(get(x))
```

Comment:
Use `unsafe_get`, and add a fast path when `isbits(T)` (similar to what happens with `null_safe_op`).